contracts/strategies/StrategyEthHegicLP.sol

```diff
@@ -48,28 +48,34 @@ contract StrategyEthHegicLP is BaseStrategy {
     // for the weth->eth swap
     receive() external payable {}
 
+    bool public withdrawFlag = false;
+
+
+    // function to designate when vault is in withdraw-state.
+    // when bool is set to false, deposits and harvests are enabled
+    // when bool is set to true, deposits and harvests are locked so the withdraw timelock can count down
+    // setting bool to one also triggers the withdraw from staking and starts the 14 day countdown
+    function setWithdrawal(bool _withdrawFlag) external {
+        require(msg.sender == strategist, "!authorized");
+        withdrawFlag = _withdrawFlag;
+        if (withdrawFlag == true) {
```

Comment:
this looks weird, usually a setter just sets the value. 
This logic should go into another method and protected to be strategist/gov only.